src/indexer/index_writer.rs

```diff
@@ -643,12 +647,45 @@ impl IndexWriter {
     pub fn add_document(&mut self, document: Document) -> u64 {
         let opstamp = self.stamper.stamp();
         let add_operation = AddOperation { opstamp, document };
-        let send_result = self.document_sender.send(add_operation);
+        let send_result = self.document_sender.send(vec![add_operation]);
         if let Err(e) = send_result {
             panic!("Failed to index document. Sending to indexing channel failed. This probably means all of the indexing threads have panicked. {:?}", e);
         }
         opstamp
     }
+
+    /// Runs a group of document operations.
```

Comment:
Document what happens if someone runs an empty `array`  of operations.

Reply:
(or `Vec<>` if we end up using `Vec`)

Reply:
We want at least 2 very simple unit test. One for when empty operations. One for non-empty.

Reply:
I added some tests, but they are only asserting that the opstamps returned by `run` are correct. There are no other assertions.

Reply:
That's fine